src/commonTest/kotlin/org/cru/godtools/tool/state/StateTest.kt

```diff
@@ -0,0 +1,51 @@
+package org.cru.godtools.tool.state
+
+import org.cru.godtools.tool.model.EVENT_NAMESPACE_STATE
+import org.cru.godtools.tool.model.EventId
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertNull
+import kotlin.test.assertTrue
+
+class StateTest {
+    private val state = State()
```

Comment:
I think I'm a little confused by what the state logic is doing/is used for. Could you explain it to me?

Reply:
it's a new concept that is being introduced into the content xml model. It will be a way for different components to interact with each other and dynamically respond to state data set by other components.

For example, a multi-select element presents several choices, when the user selects one of those choices it sets a state value for the selected choice. We can then have a button that is made visible when there is a value set to state. that button can also fire off different events based on what value was set in the state.